<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ModbusTcpListener</class>
 <widget class="QWidget" name="ModbusTcpListener">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>734</width>
    <height>528</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Modbus Tcp Server</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_3">
   <item>
    <widget class="QSplitter" name="splitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QWidget" name="layoutWidget">
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="boxConnection">
         <property name="title">
          <string>Modbus TCP Connection</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout">
          <item>
           <widget class="QLabel" name="lblConnection">
            <property name="text">
             <string>Choose TCP port to listen</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QSpinBox" name="spinPort">
              <property name="minimum">
               <number>1</number>
              </property>
              <property name="maximum">
               <number>64000</number>
              </property>
              <property name="value">
               <number>502</number>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QPushButton" name="btnConnect">
              <property name="text">
               <string>Connect</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnDisconnect">
              <property name="text">
               <string>Disconnect</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QLabel" name="lblServerId">
              <property name="text">
               <string>Server ID</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="spinServerId">
              <property name="minimum">
               <number>1</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox">
         <property name="title">
          <string>Register Editor</string>
         </property>
         <layout class="QFormLayout" name="formLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="lblTable">
            <property name="text">
             <string>Table</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cmbTable"/>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="lblValueType">
            <property name="text">
             <string>Value type</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cmbValueType"/>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="lblAddress">
            <property name="text">
             <string>Address</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="spinAddress">
            <property name="maximum">
             <number>2000</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="lblValue">
            <property name="text">
             <string>Value</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QLineEdit" name="editValue"/>
            </item>
            <item>
             <widget class="QCheckBox" name="chkValue">
              <property name="text">
               <string>On</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="4" column="1">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QPushButton" name="btnValueRead">
              <property name="text">
               <string>Read</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="btnValueWrite">
              <property name="text">
               <string>Write</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="boxAction">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Action</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_4">
          <item>
           <widget class="QComboBox" name="cmbHandler"/>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QTextBrowser" name="textLog"/>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
